File: wsh-backend/wsh-service/src/main/java/com/wsh/domain/entity/Invoice.java
package com.wsh.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 发票信息实体
 */
@Data
@TableName("tb_invoice")
public class Invoice {

    @TableId(type = IdType.ASSIGN_ID)
    private Long invoiceId;

    private Long userId;
    private Long merchantId;
    private Long consumeRecordId;

    /** 发票号码 */
    private String invoiceNo;

    /** 发票代码 */
    private String invoiceCode;

    /** 发票类型：1电子普票 2电子专票 3纸质普票 */
    private Integer invoiceType;

    /** 发票状态：0未开 1已开 2开票中 3开票失败 */
    private Integer invoiceStatus;

    /** 发票金额 */
    private BigDecimal invoiceAmount;

    /** 发票抬头 */
    private String invoiceTitle;

    /** 税号 */
    private String taxNumber;

    /** 电子发票下载链接 */
    private String invoiceUrl;

    /** 开票日期 */
    private LocalDate invoiceDate;

    /** 申请开票时间 */
    private LocalDateTime requestTime;

    /** 开票完成时间 */
    private LocalDateTime completeTime;

    /** 最近同步时间 */
    private LocalDateTime syncTime;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: wsh-miniapp/src/api/invoice.ts
import request from './request'

// 发票相关接口

export interface InvoiceItem {
  invoiceId: number
  merchantId: number
  merchantName: string
  consumeRecordId: number
  invoiceNo: string
  invoiceCode: string
  invoiceType: number
  invoiceTypeName: string
  invoiceStatus: number
  invoiceStatusName: string
  invoiceAmount: number
  invoiceTitle: string
  taxNumber: string
  invoiceUrl: string
  invoiceDate: string
  requestTime: string
  completeTime: string
  createdAt: string
}

export interface InvoiceSetting {
  settingId: number
  titleType: number
  titleTypeName: string
  invoiceTitle: string
  taxNumber: string
  bankName: string
  bankAccount: string
  companyAddress: string
  companyPhone: string
  isDefault: number
  createdAt: string
}

// 获取发票列表
export function getInvoiceList(params: { status?: number; page?: number; size?: number }) {
  return request.get<{ list: InvoiceItem[]; total: number }>('/v1/invoices', params)
}

// 获取发票详情
export function getInvoiceDetail(invoiceId: number) {
  return request.get<InvoiceItem>(`/v1/invoices/${invoiceId}`)
}

// 申请开票
export function requestInvoice(data: { consumeRecordId: number; invoiceSettingId: number }) {
  return request.post<InvoiceItem>('/v1/invoices/request', data)
}

// 获取发票抬头设置列表
export function getInvoiceSettings() {
  return request.get<InvoiceSetting[]>('/v1/invoice-settings')
}

// 新增发票抬头
export function createInvoiceSetting(data: {
  titleType: number
  invoiceTitle: string
  taxNumber?: string
  bankName?: string
  bankAccount?: string
  companyAddress?: string
  companyPhone?: string
  isDefault?: number
}) {
  return request.post<InvoiceSetting>('/v1/invoice-settings', data)
}

// 更新发票抬头
export function updateInvoiceSetting(settingId: number, data: {
  titleType?: number
  invoiceTitle?: string
  taxNumber?: string
  bankName?: string
  bankAccount?: string
  companyAddress?: string
  companyPhone?: string
}) {
  return request.put<InvoiceSetting>(`/v1/invoice-settings/${settingId}`, data)
}

// 删除发票抬头
export function deleteInvoiceSetting(settingId: number) {
  return request.delete(`/v1/invoice-settings/${settingId}`)
}

// 设为默认抬头
export function setDefaultInvoiceSetting(settingId: number) {
  return request.put(`/v1/invoice-settings/${settingId}/default`)
}
